### PR TITLE
User story23

### PR DESCRIPTION
**Pull request by KellyIB**
User stories 23-24
  added tests for user story 23 and 24
  added model tests for instance methods written in the model 
  changed cart view and controller and route

### CODE REVIEW

File: app/views/cart/show.html.erb [hunk 1]
```diff
@@ -29,13 +32,13 @@
         <p><%=number_to_currency(item.price)%></p>
       </td>
       <td>
-        <center>
-          <p><%=quantity%></p>
-          <p><%= link_to "Remove", "/cart/#{item.id}",method: :delete %></p>
-        </center>
+        <p><%= quantity %> of <%=item.name%></p>
+        <p> <%= button_to "Add One More", "/cart/#{item.id}", params: {increment_decrement: "increment"}, method: :patch %> </p>
```
Comment (adumortier) [inline app/views/cart/show.html.erb:36]: works great in rails s!

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -29,13 +32,13 @@
...
+        <p> <%= button_to "Remove One From Cart", "/cart/#{item.id}", params: {increment_decrement: "decrement"}, method: :patch %> </p>
+        <p><%= link_to "Remove", "/cart/#{item.id}",method: :delete %></p>
+        <p><%=number_to_currency(cart.subtotal(item))%></p>
         </td>
       <td>
-        <p><%=number_to_currency(cart.subtotal(item))%></p>
```
Comment (adumortier) [inline app/views/cart/show.html.erb:38]: what was this part doing?

File: config/routes.rb [hunk 3]
```diff
@@ -27,6 +27,7 @@
 
   post "/cart/:item_id", to: "cart#add_item"
   get "/cart", to: "cart#show"
+  patch "/cart/:item_id", to: "cart#increment_decrement"
```
Comment (adumortier) [inline config/routes.rb:30]: maybe we should make sure with Meg/Mike to make sure there's ok with this kind of route

File: spec/features/cart/visitor_cart_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+      visit "/cart"
+      within "#cart-item-#{@paper.id}" do
+        click_button "Add One More"
+        click_button "Add One More"
+        click_button "Add One More"
+        click_button "Add One More"
+        expect(page).to have_content("5 of #{@paper.name}")
+        click_button "Add One More"
+        expect(page).to have_content("5 of #{@paper.name}")
```
Comment (adumortier) [inline spec/features/cart/visitor_cart_spec.rb:27]: works perfectly :)

**Review by adumortier — APPROVED** [review]
Looks good to me! I think at some point we should try to address the case where a `user` tries to add an `inactive` `item` to the `cart`. At the moment, it seems that even inactive items can be added to the cart and ordered as long as the quantity ordered is lower than the inventory (unless I missed something)

File: app/controllers/cart_controller.rb [hunk 5]
```diff
@@ -26,13 +26,13 @@ def exclude_admin
...
-  #     cart.subtract_quantity(params[:item_id])
-  #     return remove_item if cart.quantity_zero?(params[:item_id])
-  #   end
-  #   redirect_to "/cart"
-  # end
+  def increment_decrement
+    if params[:increment_decrement] == "increment"
+      cart.add_quantity(params[:item_id]) unless cart.limit_reached?(params[:item_id])
+    elsif params[:increment_decrement] == "decrement"
+      cart.subtract_quantity(params[:item_id])
```
Comment (cgaddis36) [inline app/controllers/cart_controller.rb:33]: Logic looks good here I like the use of passing in the string in the params!

File: spec/features/cart/visitor_cart_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      end
+    end
+
+    it "I can decrement the amount of an item in my cart. when cart quantity is zero, item is removed from cart" do
+      visit "/items/#{@pen.id}"
+      click_on "Add To Cart"
+
+      visit "/items/#{@pencil.id}"
+      click_on "Add To Cart"
```
Comment (cgaddis36) [inline spec/features/cart/visitor_cart_spec.rb:36]: Solid tests, all passing on my end.

**Review by cgaddis36 — APPROVED** [review]

**Pull request merged**